sqlmesh/core/plan/evaluator.py

```diff
@@ -376,3 +378,13 @@ def can_evaluate_before_promote(
     return not snapshot.is_paused_forward_only and not any(
         snapshots[p_id].is_paused_forward_only for p_id in snapshot.parents
     )
+
+
+def update_intervals_for_new_snapshots(
```

Comment:
why is this done in the plan evaluator?

Reply:
Because it's a part of the plan evaluation. Where should it be instead?